Collector

kangwork opened the pull request:
**Description**
This pull request implements the specifications discussed in the meeting on June 6th, 2024.

**Key Changes**
- Implemented the main features as instructed.
- Note: This implementation has not yet been thoroughly tested.

**Next Steps**
- Complete thorough testing of the changes.
- Rebuild all GCP resources (instances) for testing purposes.

**Review Plan**
- Assign mentors as reviewers on Monday after testing is complete.

**Additional Notes**
- Please be aware that this is a work in progress and subject to further revisions based on test results. 🥹

kangwork committed:
Wrote some comments and a new function to get details of a bucket with a given name
Need to be tested and revised on a company laptop

kangwork committed:
Completed get_storage_buckets.py except the auto deployment part

kangwork committed:
Commited the part undone

kangwork committed:
Completed testing of get_iam_roles.py

kangwork committed:
Replaced the print statements to the use of a python's logging package,
Added some comments that explain the fact that the global variable 'default_project_id' serves as an example default, but the user can specify the id by giving a query parameter, and
Changed the variable name to be more intuitive (i.e. project_id --> default_project_id)

kangwork committed:
Added logger.py,
Made the get_iam_roles.py log the output using the logging package

kangwork committed:
Fixed a minor newline typo

kangwork committed:
backup

kangwork committed:
Merging all the sub-branches for consistency

kangwork committed:
Synced the .gitignore file and removed some unnecessary lines in logging.py

kangwork committed:
Reorganized the directories & files

kangwork committed:
Added main.py that calls the resources collector (Not yet tested + Need to classify+rename the methods)

kangwork committed:
A test commit

kangwork committed:
A test commit 2

kangwork committed:
Fixed logging.py so that it sets up its  file/stream handler interactively later

kangwork committed:
Fixed imports

kangwork committed:
Replaced the use of the hardcoded file path to the use of environment variables

kangwork committed:
Changed the get_credentials feature as suggested in the meeting last night
- Instead of getting the project_id separately, find it from the credential file
- The credential data is also not hardcoded, but yieldable through accessing an  environment variable

kangwork committed:
Removed unnecessary lines in main.py

kangwork committed:
Removed comments as much as possible
Added an error handling (a case that the env variable is not set)

kangwork committed:
Removed all the manual project retrieval

kangwork committed:
Updated the comment and the instruction on the welcome page

kangwork committed:
Made each resources classes

kangwork committed:
Function name, calls, variables, and classes consistency

kangwork committed:
Updated the IAM roles related routing messages on the welcomepage

kangwork committed:
Made the setup_logger calls consistent
:

kangwork committed:
Updated logging setup + makefile

kangwork committed:
Complete the main.py testing

kangwork committed:
Updated logging (could not resolve the first duplicate starting messages- but will resolve it) and the READme file

yjinjo commented on src/main.py:
```diff
@@ -4,7 +4,7 @@
 from google.cloud import storage
 from utils.credentials import get_credentials
```
쓰지 않는 라이브러리나 모듈이 있다면 `import` 하지 않아도 됩니다.

kangwork replied:
앗 확인하고 모두 수정하겠습니다! 감사합니다!

yjinjo commented on src/main.py:
```diff
@@ -18,35 +18,41 @@
-    return templates.TemplateResponse("template.html", {"request": request, "json_data": json_data})
-    # return {"data": message}
+    try:
+        message += storage_buckets.get_route_messages(default_project_id)
+        message += iam_roles.get_route_messages(default_project_id)
+        message += vm_instances.get_route_messages(default_project_id)
```
`message` 가 중복되어 보이는데, 아래와 같이 코드를 작성하거나 함수로 따로 작성하시는게 어떨까요?

```python
route_messages = [
    storage_buckets.get_route_messages(default_project_id),
    iam_roles.get_route_messages(default_project_id),
    vm_instances.get_route_messages(default_project_id)
]
message += "".join(route_messages)
```

kangwork replied:
넵 좋은 것 같아요. 계획해보고 말씀드리도록 하겠습니다!

yjinjo commented on src/main.py:
```diff
@@ -18,35 +18,41 @@
     message = f"""Welcome to the Mini Google Cloud Collector!
     
         Avaialable Routes:
         """
```
expression이 없다면 굳이 f-string을 안쓰셔도 됩니다.

kangwork replied:
넵 수정하겠습니다! 감사합니다.^^

kangwork committed:
Added a readme comment 'navigate to the source directory'

kangwork committed:
Updated the makefile

yjinjo reviewed (approved):
LGTM 👍

kangwork committed:
- Removed unnecessary imports
- Removed unnecessary f-string
- Made the get_route_messages function methods as Minho Kim suggested

kangwork committed:
Changed the resource class init for flexibility

kangwork committed:
Removed an unused template

kangwork committed:
Added collector classes

kangwork committed:
Replaced the collect_resource(s) calls to OOP based calls using collector objects

kangwork committed:
get_route_messages inheritance

kangwork committed:
Removed unnecessary comments in Makefile

kangwork committed:
Added decorators and updated error handling in subfiles (not yet implemented decorators in main.py)

kangwork committed:
Update minor code styling in main.py

kangwork committed:
Made the get_credential function to be called once

kangwork committed:
Removed an unnecessary import

kangwork committed:
Updated README + Makefile and removed an unnecessary prompt in main.py

kangwork committed:
[Not completed] Fixing Makefile so that it works as the README file instructs
